var/spack/repos/builtin/packages/nvshmem/package.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+from spack import *
+import re
+
+
+class Nvshmem(MakefilePackage, CudaPackage):
+    """NVSHMEM is a parallel programming interface based on OpenSHMEM that
+    provides efficient and scalable communication for NVIDIA GPU
+    clusters. NVSHMEM creates a global address space for data that spans
+    the memory of multiple GPUs and can be accessed with fine-grained
+    GPU-initiated operations, CPU-initiated operations, and operations on
+    CUDA streams."""
+
+    homepage = "https://developer.nvidia.com/nvshmem"
+    url      = "https://developer.nvidia.com/nvshmem-src-203-0"
+
+    maintainers = ['bvanessen']
+
+    version('2.0.3-0', sha256='20da93e8508511e21aaab1863cb4c372a3bec02307b932144a7d757ea5a1bad2', extension='txz')
+
+    variant('cuda', default=True, description='Build with CUDA')
```

Comment:
Is this meant to express it's cuda only? In that case
```suggestion
    conflicts('~cuda')
```
Redefining the variant is probably not a good idea.